scripts/version.mjs

```diff
@@ -0,0 +1,130 @@
+import { join } from 'path';
+import semver from 'semver';
+import { promisify } from 'util';
+
+import { getPackages, toPackageNamesMap } from './utils.mjs';
+
+const { black, bold, dim } = chalk;
+const { writeFile } = promises;
+const { minVersion } = semver;
+const exec = promisify(childProcess.exec);
+
+const versioningText = black.bgYellow(' VERSIONING ');
+const updatingText = black.bgYellow('  UPDATING  ');
+const skippedText = black.bgWhite('  SKIPPING  ');
+const failedText = black.bgRed('   FAILED   ');
+const completeText = black.bgGreen('  COMPLETE  ');
+
+const updateVersion = async version => {
+  console.log(versioningText, bold('workspaces'), dim(version));
+  // Run the default version command for all workspaces
+  await exec(`npm version ${version} --workspaces`);
+  console.log(completeText, bold('workspaces'), dim(version));
+};
+
+const getNextPackageJson = (packageJson, packagesByName) => {
+  if (
+    !packageJson.dependencies &&
+    !packageJson.devDependencies &&
+    !packageJson.peerDependencies
+  ) {
+    return;
+  }
+
+  let nextPackageJson;
+  const dependencyKeys = [
+    'dependencies',
+    'devDependencies',
+    'peerDependencies',
+  ];
+
+  for (const dependencyKey of dependencyKeys) {
+    for (const [dependency, versionRange] of Object.entries(
+      packageJson[dependencyKey] ?? {}
+    )) {
+      if (!packagesByName.has(dependency)) {
+        continue;
+      }
+
+      const { version } = minVersion(versionRange);
+      if (!version) {
+        throw new Error('Unable to find version for range', versionRange);
+      }
+
+      const [, { version: nextVersion }] = packagesByName.get(dependency);
+
+      if (!nextPackageJson) {
+        nextPackageJson = Object.assign({}, packageJson);
+      }
+
+      nextPackageJson[dependencyKey][dependency] = versionRange.replace(
+        version,
+        nextVersion
+      );
+    }
+  }
+
+  return nextPackageJson;
+};
+
+const updatePackageDependencies = async packages => {
+  const packagesByName = toPackageNamesMap(packages);
+
+  let i = 0;
+  for (const [path, packageJson] of packages) {
+    const { name } = packageJson;
+    const details = [bold(name), dim('package.json')];
+
+    console.log(updatingText, ...details);
+
+    try {
+      const nextPackageJson = getNextPackageJson(packageJson, packagesByName);
+      if (nextPackageJson) {
+        await writeFile(
+          join(path, 'package.json'),
+          `${JSON.stringify(nextPackageJson, null, 2)}\n`
+        );
+        console.log(completeText, ...details);
+      } else {
+        console.log(skippedText, ...details);
+      }
+    } catch (err) {
+      console.log(failedText, ...details);
+      console.log();
+
+      throw err;
+    }
+
+    if (i < packages.size - 1) {
+      console.log();
+    }
+
+    i += 1;
+  }
+};
+
+const updatePackageLockJson = async () => {
+  console.log(updatingText, bold('package-lock.json'));
+  await exec('npm install');
```

Comment:
I'm not sure if we should update `package-lock.json` while bumping versions, I would suggest to run `npm ci` instead.

Reply:
This uses the same process we have today, where we version and commit locally, then push up the version tag so that the `publish` script activates. Thus, there is no need to use `npm ci` here since it is performed locally.